README.md

```diff
@@ -17,43 +29,56 @@ There are functions for KZG operations:
+executed once during the initialization process. As the name suggests, the
+[trusted setup
+file](https://github.com/ethereum/c-kzg-4844/blob/main/src/trusted_setup.txt) is
+considered to be trustworthy.
 
 - `load_trusted_setup`
 - `load_trusted_setup_file`
 - `free_trusted_setup`
 
-## Bindings
+## Remarks
 
-There are bindings for the following languages:
+### Tests
 
-| Language | Link                                 |
-|----------|--------------------------------------|
-| C#       | [README](bindings/csharp/README.md)  |
-| Go       | [README](bindings/go/README.md)      |
-| Java     | [README](bindings/java/README.md)    |
-| Nim      | [README](bindings/nim/README.md)     |
-| Node.js  | [README](bindings/node.js/README.md) |
-| Python   | [README](bindings/python/README.md)  |
-| Rust     | [README](bindings/rust/README.md)    |
+All the bindings are tested against the [KZG reference
+tests](https://github.com/ethereum/consensus-spec-tests/tree/master/tests/general/deneb/kzg),
+which are defined in the consensus-spec-tests. Additionally, a suite of unit
+tests for internal C functions is located
+[here](https://github.com/ethereum/c-kzg-4844/blob/main/src/test_c_kzg_4844.c).
+
+### Parallelization
 
-## Installation
+The interface functions in C-KZG-4844 are single-threaded for simplicity, as
+implementing multi-threading across multiple platforms can be complex. While
+performance is important, these functions are already quite fast and efficient.
+For instance, `verify_blob_kzg_proof` is expected to finish in under 3ms on most
+systems.
 
-### Prerequisites
+### Batched verification
 
-The following must be installed:
+When processing multiple blobs, `verify_blob_kzg_proof_batch` is more efficient
+than calling `verify_blob_kzg_proof` individually. In CI tests, verifying 64
+blobs in batch is 53% faster per blob than verifying them individually. For a
+single blob, `verify_blob_kzg_proof_batch` calls `verify_blob_kzg_proof`, and
+the overhead is negligible.
 
-* `git`
-* `make`
-* `clang`
+### Benchmarks
 
-### Build & test
+C-KZG-4844 does not include C benchmarks; however, some bindings (Go, Java, and
+Rust) have their own benchmarks. Including benchmarks in the bindings offers a
+more realistic performance estimate, as C-KZG-4844 is not expected to be used
+outside of the bindings.
 
-To build `c_kzg_4844.o`, the object file that the bindings use, run `make` in
-the `src` directory. This will ensure the `blst` submodule has been initialized,
-build `blst`, build `c_kzg_4844`, and run the tests. From the project root, run
-this:
+### Why C?
 
-```
-cd src && make
-```
+The primary reason for choosing C is because
+[blst](https://github.com/supranational/blst), the BLS12-381 signature library
+we wanted to use, is mostly written in C. We were concerned that using blst with
+another language, like Rust, and then building bindings on top of it would
+introduce too much overhead.  Furthermore, the C toolchain is ubiquitous, and it
```

Comment:
Given that the FFI overhead from rust is not really a concern anymore, perhaps you can remove the overhead sentence. Another valid positive of doing it in C is that it doesn't have the heavy Rust runtime (smaller memory/binary footprint).

Reply:
Will update this paragraph to reflect this 👍